tensorflow/compiler/mlir/hlo/include/mlir-hlo/Dialect/mhlo/IR/hlo_ops.td

```diff
@@ -2162,4 +2162,124 @@ def HLO_ReducePrecisionOp :
   let results = (outs HLO_FpTensor:$output);
 }
 
+def HLO_H2DOp: HLO_Op<"h2d", [NoSideEffect, SameOperandsAndResultType]> {
+  let summary = "H2D operator";
+  let description = [{
+    Copy `operand` from host to device.
```

Comment:
This is a bit terse and does not provide much more information than the "summary".

What I'm looking for here is a better semantic description. Contrary to mathematical operator which are pure and fairly well defined in general (we can almost always points to a TensorFlow/Numpy/XLA documentation to describe their behavior), this has some runtime integration assumption.
For example: are these blocking call in the generated code? 
Can we reorder two `h2d` operations? There is no attribute to distinguish between them so I assume the answer is "no". That means that we have to preserve ordering of these in a module. However you marked these as "NoSideEffect" which makes me wonder what prevents the compiler to reorder them?
There are other questions around this in control flow for example. What happens if there is a mismatch between the host and the generated code? If the answer is just that this is undefined behavior if these is a mismatch, then make it explicit.

Don't hesitate to add code samples in the op descriptions in general if it helps clarifying.

Reply:
Something else that we can consider: if these ops are specific to the RAL abstraction, we could pull them in a separate dedicated dialect like `mhlo_ral` for example.

Reply:
Right. This advice is good for us.